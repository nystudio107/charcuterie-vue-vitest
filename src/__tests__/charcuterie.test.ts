import {getAllItems, getItemsOfType, addCharcuterieItem, CharcuterieItemType} from '../index';
import {test, expect, expectTypeOf} from 'vitest';

test('Charcuterie suite', () => {
  addCharcuterieItem({
    name: 'olives',
    type: CharcuterieItemType.Fruit,
    calories: 100
  });
  addCharcuterieItem({
    name: 'salami',
    type: CharcuterieItemType.Meat,
    calories: 200
  });
  const fruits = getItemsOfType(CharcuterieItemType.Fruit);
  fruits.forEach((value: CharcuterieItem) => {
    expectTypeOf(value.type).toEqualTypeOf<CharcuterieItemType.Fruit>();
  })
  expect(getAllItems()).toMatchSnapshot();
})
