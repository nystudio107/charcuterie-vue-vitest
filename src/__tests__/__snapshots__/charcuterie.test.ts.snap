// Vitest Snapshot v1

exports[`Charcuterie suite 1`] = `
[
  {
    "calories": 100,
    "name": "olives",
    "type": 2,
  },
  {
    "calories": 200,
    "name": "salami",
    "type": 1,
  },
]
`;
